Assets/Scripts/InventoryAndItemsScripts/Shop/ShopController.cs

```diff
@@ -0,0 +1,266 @@
+using GameStateManagement;
+using Inventory;
+using Items;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Interactions;
+using UnityEngine.UI;
+using UnityEngine.UIElements;
+using static UnityEngine.InputSystem.InputAction;
+
+public class ShopController : MonoBehaviour
+{
+    public List<ItemData> shopItems; // place items to be displayed in shop
+
+    #region Keybinds
+    [SerializeField]
+    [Tooltip("The keybind that controls opening and closing the inventory.")]
+    InputAction InventoryAction;
+
+    [SerializeField]
+    [Tooltip("The keybind for interacting with items in the inventory.")]
+    InputAction SelectItem;
+
+    // Keybinds that control navigating the items within the inventory.
+    [SerializeField]
+    [Tooltip("The keybind that controls moving left in the inventory.")]
+    InputAction NavLeft;
+
+    [SerializeField]
+    [Tooltip("The keybind that controls moving right in the inventory.")]
+    InputAction NavRight;
+
+    [SerializeField]
+    [Tooltip("The keybind that controls moving up in the inventory.")]
+    InputAction NavUp;
+
+    [SerializeField]
+    [Tooltip("The keybind that controls moving down in the inventory.")]
+    InputAction NavDown;
+    #endregion
+
+    [SerializeField]
+    private GameObject invObj; // canvas game object
+
+    [SerializeField]
+    private GameObject playerInvObj; // player's inventory
+
+    // Canvas children
+    private GameObject invPanel; // main inventory panel within canvas
+    private GameObject playerCoins; // Player coins
+    private GameObject coins; // coin UI object in shop
+    private GameObject itemInfo; // item name & description parent (holds the two fields below)
+    private GameObject itemName;
+    private GameObject itemDescription;
+    private GameObject itemPrice;
+
+    private PlayerInventory pInv;
+    private ShopInventory shopInv;
+
+    private List<GameObject> inventorySlots; // slots in the inventory 
+    private List<ItemData> itemsInInventory; // list of items currently held in inventory
+    private int inventoryIndex = 0;
+    private int inventoryRowSize;
+    private UnityEngine.UI.Image selectedSlot;
+
+    public GameObject GetShopInventoryPanel() { return invObj; }
+
+    private void Start()
+    {
+        pInv = PlayerInventory.Instance;
+        shopInv = ShopInventory.Instance;
+        invPanel = invObj.transform.GetChild(0).gameObject;
+
+        playerInvObj.SetActive(true);
+        playerCoins = GameObject.FindWithTag("Coins"); // get coins from player
+        playerInvObj.SetActive(false);
+
+        coins = invObj.transform.GetChild(1).gameObject.transform.GetChild(1).gameObject;
```

Comment:
probably a better way to go about this, but I think it works fine if you're setting up every shop in the exact same way every time